book/src/design/proving-system/vanishing.md

```diff
@@ -0,0 +1,73 @@
+# Vanishing argument
+
+Having committed to the circuit assignments, the prover now needs to demonstrate that the
+various circuit relations are satisfied:
+
+- The custom gates, represented by polynomials $\text{gate}_i(X)$.
+- The rules of the lookup arguments.
+- The rules of the equality constraint permutations.
+
+Each of these relations is represented as a polynomial of degree $d$ (the maximum degree
+of any of the relations).
+
+> In our [example](../proving-system.md#example), these would be the gate polynomials:
+>
+> - $\text{gate}_0(X) = a_0(X) \cdot a_1(X) \cdot a_2(X \omega^{-1}) - a_3(X)$
+> - $\text{gate}_1(X) = f_0(X \omega^{-1}) \cdot a_2(X)$
+> - $\text{gate}_2(X) = f_0(X) \cdot a_3(X) \cdot a_0(X)$
+
+The simple way to achieve this is to divide each polynomial relation by the vanishing
+polynomial $t(X) = (X^{n} - 1)$, which is the lowest-degree monomial that has roots at
+every $\omega^i$. If any $\text{gate}_i(X)$ is perfectly divisible by $t(X)$, it is equal
+to zero over the domain (as desired).
+
+This simple construction would require a polynomial commitment per relation. Instead, we
+commit to all of the circuit relations simultaneously: the verifier samples $y$, and then
+the prover constructs the quotient polynomial
+
+$$h(X) = \frac{\text{gate}_0(X) + y \cdot \text{gate}_1(X) + \dots + y^i \cdot \text{gate}_i(X) + \dots}{t(X)},$$
+
+as a random (the prover commits to the cell assignments before the verifier samples $y$)
+linear combination of the circuit relations.
+
+- If this random linear combination is perfectly divisible by $t(X)$, then with high
+  probability all relations are satisfied.
+- Conversely, if at least one relation is not satisfied, then with high probability the
+  random linear combination of relations is not perfectly divisible by $t(X)$.
```

Comment:
It's with high probability that h(x) t(x) would not equal the evaluation of the numerator at x since this cannot hold for most values of x in the field. The probability that the _polynomial_ (in formal indeterminate X) is not perfectly divisible would in this case be 100%.